google/cloud/pubsub/testing/mock_message_batch.h

```diff
@@ -17,6 +17,7 @@
 
 #include "google/cloud/pubsub/internal/message_batch.h"
 #include "google/cloud/pubsub/message.h"
+#include "google/cloud/future_void.h"
```

Comment:
I would argue the base class header *must* define `future<void>`, and `std::functional` for you.  But if you need to appease the IWYU tooling, then this is missing `#include <functional>` too.

Reply:
Done